dvc/command/experiments.py

```diff
@@ -194,10 +215,19 @@ def _collect_rows(
             _format_time(exp.get("timestamp")),
             parent,
         ]
+        metric_names_filled = _fill_empty_values(
+            error_rev, metric_names, onerror=onerror
+        )
+        param_names_filled = _fill_empty_values(
+            error_rev, param_names, onerror=onerror
+        )
```

Comment:
If we move the error handling into `experiments.show()` we could get rid of these helpers and simplify things here. (see my other comment)

W/the moved error handling, we can just pass the a fill character into `_extend_row` like
```
_extend_row(..., fill_value="!" if exp.get("errors") else "-")
```

---

I get that these helpers are trying to display as much information as possible for everything that we were able to successfully parse for a particular rev, but it seems like an unnecessary level of complexity to me.

(Without the helpers we will still display whatever metrics/params we were able to successfully collect, the only difference is that `-` or `!` will be used in every other empty column depending on whether or not any error occurred for that rev, rather than having `!` in specific columns)